source/extensions/filters/http/common/aws/signer_impl.cc

```diff
@@ -0,0 +1,256 @@
+#include "extensions/filters/http/common/aws/signer_impl.h"
+
+#include "envoy/common/exception.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/assert.h"
+#include "common/common/hex.h"
+#include "common/common/stack_array.h"
+#include "common/http/headers.h"
+
+#include "openssl/evp.h"
+#include "openssl/hmac.h"
+#include "openssl/sha.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Common {
+namespace Aws {
+
+static const std::string AWS4{"AWS4"};
+static const std::string AWS4_HMAC_SHA256{"AWS4-HMAC-SHA256"};
+static const std::string AWS4_REQUEST{"aws4_request"};
+static const std::string CREDENTIAL{"Credential"};
+static const std::string SIGNED_HEADERS{"SignedHeaders"};
+static const std::string SIGNATURE{"Signature"};
+static const std::string HASHED_EMPTY_STRING{
```

Comment:
We don't allow non-POD statics in Envoy due to the static initialization fiasco. You might want to look at the `ConstSingleton` pattern for these, or the `CONSTRUCT_ON_FIRST_USE` style. There's a few ways this is handled. Also, constants should be naed `SignedHeaders` etc.

Reply:
@htuch shall we fail on CI for this? clang-tidy can check non-POD statics.

Also I think we should modernize some commonly used interface like HeaderMap to take `string_view` directly, `string_view` is constexpr constructor so they are safe to constructed as static.

Reply:
https://www.youtube.com/watch?v=9_th4LsV9kE